sycl/source/detail/memory_manager.cpp

```diff
@@ -1692,21 +1692,31 @@ void MemoryManager::ext_oneapi_fill_cmd_buffer(
     unsigned int ElementSize,
     std::vector<sycl::detail::pi::PiExtSyncPoint> Deps,
     sycl::detail::pi::PiExtSyncPoint *OutSyncPoint) {
-  (void)Size;
   assert(SYCLMemObj && "The SYCLMemObj is nullptr");
 
   const PluginPtr &Plugin = Context->getPlugin();
   if (SYCLMemObj->getType() != detail::SYCLMemObjI::MemObjType::Buffer) {
     throw sycl::exception(sycl::make_error_code(sycl::errc::invalid),
                           "Images are not supported in Graphs");
   }
-  if (Dim <= 1) {
+
+  // 2D and 3D buffers accessors can't have custom range or the data will
+  // likely be discontiguous.
+  bool RangesUsable = (Dim <= 1) || (Size == AccessRange);
+  // For 2D and 3D buffers, the offset must be 0, or the data will be
+  // discontiguous.
+  bool OffsetUsable = (Dim <= 1) || (AccessOffset == sycl::id<3>{0, 0, 0});
+  size_t RangeMultiplier = AccessRange[0] * AccessRange[1] * AccessRange[2];
+
+  if (RangesUsable && OffsetUsable) {
```

Comment:
Thanks for this feedback. According to my understanding, this exception/error  is triggered if users try to perform a fill op with invalid accessor dimensions or offsets. So. since users can trigger this error, we prefer to keep it as an exception rather than an assert.

Reply:
Alright. The comment on 1718-1719 read to me like it shouldn't be reachable by the user. A slight rephrasing wouldn't hurt, but I don't think it has to be a blocker for this.